threat_intel/virustotal.py

```diff
@@ -9,174 +9,241 @@
 
 
 class VirusTotalApi(object):
-    BASE_DOMAIN = u'https://www.virustotal.com/vtapi/v2/'
+    BASE_DOMAIN = u'https://www.virustotal.com/api/v3/'
 
-    def __init__(self, api_key, resources_per_req=25, cache_file_name=None, update_cache=True, req_timeout=None):
+    def __init__(self, api_key, cache_file_name=None, update_cache=True, req_timeout=None):
         """Establishes basic HTTP params and loads a cache.
 
         Args:
             api_key: VirusTotal API key
-            resources_per_req: Maximum number of resources (hashes, URLs)
-                to be send in a single request
             cache_file_name: String file name of cache.
             update_cache: Determines whether cache should be written out back to the disk when closing it.
                           Default is `True`.
             req_timeout: Maximum number of seconds to wait without reading a response byte before deciding an error has occurred.
                          Default is None.
         """
-        self._api_key = api_key
-        self._resources_per_req = resources_per_req
-        self._requests = MultiRequest(req_timeout=req_timeout)
+        self._requests = MultiRequest(req_timeout=req_timeout, default_headers={'x-apikey': api_key})
 
         # Create an ApiCache if instructed to
         self._cache = ApiCache(cache_file_name, update_cache) if cache_file_name else None
 
     @MultiRequest.error_handling
-    def get_file_reports(self, resources):
+    def get_file_reports(self, file_hash_list):
         """Retrieves the most recent reports for a set of md5, sha1, and/or sha2 hashes.
 
         Args:
-            resources: list of string hashes.
+            file_hash_list: list of string hashes.
         Returns:
             A dict with the hash as key and the VT report as value.
         """
         api_name = 'virustotal-file-reports'
+        api_endpoint = 'files/{}'
 
-        all_responses, resources = self._bulk_cache_lookup(api_name, resources)
-        resource_chunks = self._prepare_resource_chunks(resources)
-        response_chunks = self._request_reports("resource", resource_chunks, 'file/report')
+        all_responses, file_hash_list = self._bulk_cache_lookup(api_name, file_hash_list)
+        response_chunks = self._request_reports(file_hash_list, api_endpoint)
         self._extract_response_chunks(all_responses, response_chunks, api_name)
```

Comment:
If my eyes are not failing me, this code block and the similar ones in the other methods could be replaced by a calls to `_extract_all_responses`

Reply:
your eyes are 20/20